tests/deprecated_api/__init__.py

```diff
@@ -13,9 +13,27 @@
 # limitations under the License.
 """Test deprecated functionality which will be removed in vX.Y.Z"""
 import sys
+from contextlib import contextmanager
+from typing import Optional
+
+import pytest
 
 
 def _soft_unimport_module(str_module):
     # once the module is imported  e.g with parsing with pytest it lives in memory
     if str_module in sys.modules:
         del sys.modules[str_module]
+
+
+@contextmanager
+def no_deprecated_call(match: Optional[str] = None):
```

Comment:
can we use this? https://github.com/PyTorchLightning/pytorch-lightning/blob/a3310cf62677d75b76851fa3eb33a5cf9468090d/tests/helpers/utils.py#L121
you suggested it on other PR, I just modified it to support other warning types too.

Reply:
It's not merged yet, right? Will do a follow-up PR for the consistency.

Reply:
there are high chances that PR will be changed and won't be having this function in the end. so you can still add it if you want. In the end it will just generate a conflict which can be resolved easily in case that one gets merged with this.

Reply:
Oh, you meant to add the function here itself. Cool!

Reply:
I love how a utility function is moving from one PR to another 😂

Reply:
ingenious move :stuck_out_tongue_closed_eyes: